packages/core-data/src/selectors.ts

```diff
@@ -321,7 +321,7 @@ export interface GetEntityRecord {
 	>(
 		kind: string,
 		name: string,
-		key: EntityRecordKey,
+		key: EntityRecordKey | undefined,
```

Comment:
This seems to work, and I'm not a TypeScript expert so others may want to chime in. However, I'd expect the `GetEntityRecord` interface to declare the key as optional as well (as opposed to allow `undefined` as a value).

Reply:
I added this originally as it was causing TS errors in the TS version of the action, and setting the `key` as optional didn't seem to work. However, I've tried setting it to optional again [here](https://github.com/WordPress/gutenberg/pull/65426/commits/b2e88eb33715f590693a9211623e3c0117815443) and I'm not seeing any TS errors, so fingers crossed this works fine.